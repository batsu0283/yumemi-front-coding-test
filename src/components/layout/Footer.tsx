export const Footer = () => (
  <footer className="bg-gray-800 text-white p-4 mt-10 text-center">
    <p>&copy; batsu0283</p>
  </footer>
);
