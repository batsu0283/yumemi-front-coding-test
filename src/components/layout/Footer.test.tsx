import { render, screen } from "@testing-library/react";
import { Footer } from "./Footer";

describe("Footer", () => {
  it("指定されたタイトルが表示されること", () => {
    const testTitle = "フッターのテストタイトル";
    render(<Footer title={testTitle} />);
    expect(screen.getByText(testTitle)).toBeInTheDocument();
  });

  it("footerの要素が存在すること", () => {
    render(<Footer title="フッタータイトル" />);
    expect(screen.getByRole("contentinfo")).toBeInTheDocument();
  });

  it("Snapshot", () => {
    const { container } = render(<Footer title="タイトル" />);
    expect(container).toMatchSnapshot();
  });
});
