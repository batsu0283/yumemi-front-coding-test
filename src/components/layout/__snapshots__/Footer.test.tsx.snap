// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer > Snapshot 1`] = `
<div>
  <footer
    class="bg-gray-800 text-white p-4 mt-10 text-center"
  >
    <p>
      タイトル
    </p>
  </footer>
</div>
`;
